docs/guides/modding-guide.rst

```diff
@@ -0,0 +1,81 @@
+.. _modding-guide:
+
+DFHack modding guide
+====================
+
+Preface
```

Comment:
I think this might need a little reorganization. We should start out with an explicit "What is a mod?" section to define terms and make clear who the intended audience is for this guide.

The section can go over the difference between a script (something you "run" to make a programmatic change to game state) and a mod (something you "install" to get a desired persistent change to game content or behavior) and the relationship between the two 
(e.g. how and why are scripts used by mods)

Then the next section can maybe show an actual comparison between how something complicated to do just in the raws is made simpler by combining raw mods with scripts. Like, actually show the raw-only implementation and then show the raw plus script implementation. Even if people don't know anything about raws or scripts, this example should visually show that the script method is simpler. This will be your "hook" and the reason why people will want to read the rest of the document.